bugsnag.php

```diff
@@ -133,11 +147,37 @@ public function initActions()
 
     public function adminMenuActions()
     {
-        // Add the "settings" link to the Bugsnag row of plugins.php
-        add_filter('plugin_action_links', array($this, 'pluginActionLinksFilter'), 10, 2);
+        if ( ! function_exists( 'is_plugin_active_for_network' ) || ! is_plugin_active_for_network($this->pluginBase)) {
+            // Add the "settings" link to the Bugsnag row of plugins.php
+            add_filter('plugin_action_links', array($this, 'pluginActionLinksFilter'), 10, 2);
+
+            // Create the settings page
+            add_options_page('Bugsnag Settings', 'Bugsnag', 'manage_options', 'bugsnag', array($this, 'renderSettings'));
+        }
+    }
+
+    public function networkAdminMenuActions()
+    {
+        if (function_exists('is_plugin_active_for_network') && is_plugin_active_for_network($this->pluginBase)) {
+            // Create the network settings page
+            add_submenu_page('settings.php', 'Bugsnag Settings', 'Bugsnag', 'manage_network_options', 'bugsnag', array($this, 'renderSettings'));
+        }
+    }
 
-        // Create the settings page
-        add_options_page('Bugsnag Settings', 'Bugsnag', 'manage_options', 'bugsnag', array($this, 'renderSettings'));
+    private function updateNetworkSettings( $settings )
+    {
+        // Update options
+        update_site_option('bugsnag_api_key', isset($_POST['bugsnag_api_key']) ? $_POST['bugsnag_api_key'] : '');
+        update_site_option('bugsnag_notify_severities', isset($_POST['bugsnag_notify_severities']) ? $_POST['bugsnag_notify_severities'] : '');
+        update_site_option('bugsnag_filterfields', isset($_POST['bugsnag_filterfields']) ? $_POST['bugsnag_filterfields'] : '');
+        update_site_option('bugsnag_network', true);
+
+        // Update variables
```

Comment:
Note that this code exists due to the variables already being defined before the new options are saved. This was forcing the settings page to retain the old values for the inputs until the page was refreshed.